backend/services/inventory/tests/tests/test_inventory_searching.py

```diff
@@ -11,55 +11,53 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
-from common import (
-    inventory_attributes,
-    management_client,
-    internal_client,
-    clean_db,
-    mongo,
-)
 
-import requests
-import pytest
+import json
 import os
 
-from client import DEFAULT_AUTH
+import pytest
+
+import openapi_client as oas
+
+from client import make_authenticated_client
 
 
 @pytest.mark.usefixtures("clean_db")
 class TestInventorySearching:
-    def test_inventory_searching(
-        self, management_client, internal_client, inventory_attributes
-    ):
+    def test_inventory_searching(self, inventory_attributes):
+        api_client = make_authenticated_client(is_device=False)
+        internal_client = oas.InventoryInternalApi()
         extra_inventory_items = {
             "users_logged_in": 100,
             "open_connections": 1231,
             "open_ports": 523,
         }
 
-        for i in extra_inventory_items.keys():
+        for key in extra_inventory_items.keys():
             it = list(inventory_attributes)
             it.append(
-                management_client.inventoryAttribute(
-                    name=i, value=extra_inventory_items[i]
+                oas.Attribute(
+                    name=key, value=oas.AttributeValue(extra_inventory_items[key])
                 )
             )
 
             did = "".join([format(i, "02x") for i in os.urandom(128)])
-            internal_client.create_device(did, it)
-
-        r = requests.get(
-            management_client.client.swagger_spec.api_url + "/devices",
-            params=({"users_logged_in": 100}),
-            verify=False,
-            headers={"Authorization":DEFAULT_AUTH},
-        )
-        assert len(r.json()) == 1
+            internal_client.initialize_device(
+                tenant_id="", device_new=oas.DeviceNew(id=did, attributes=it)
+            )
 
-        r = requests.get(
-            management_client.client.swagger_spec.api_url + "/devices",
-            params=({"open_connections": 1231}),
-            verify=False,
-            headers={"Authorization":DEFAULT_AUTH},
-        )
-        assert len(r.json()) == 1
+        # HACK: Because of the API parameters for searching is not proper,
+        # we have to interact with the api_client interface directly.
+        for q in [{"users_logged_in": 100}, {"open_connections": 1231}]:
+            req = api_client.param_serialize(
+                "GET",
+                "/api/management/v1/inventory/devices",
+                query_params=q,
+                header_params={
+                    "Authorization": f"Bearer {api_client.configuration.access_token}"
+                },
+            )
+            rsp = api_client.call_api(*req)
```

Comment:
okey, why are we doing this? this defies all idea behind porting to open api 3 and this task. why do we have to do it? could we create a follow up task to fix it?

Reply:
I'm not sure how to fix this, do you have a suggestion?
We could provide an alternative parameterization that would be compatible with both OAS and the generator.